visualization/app/codeCharta/state/effects/updateFileSettings/updateFileSettings.effect.ts

```diff
@@ -25,15 +24,21 @@ export class UpdateFileSettingsEffect {
 				const visibleFiles = getVisibleFiles(state.files)
 				const withUpdatedPath = isPartialState(state.files)
 				const allAttributeTypes = visibleFileStatesSelector(state).map(({ file }) => file.settings.fileSettings.attributeTypes)
+				const allAttributeDescriptors = visibleFileStatesSelector(state).map(
+					({ file }) => file.settings.fileSettings.attributeDescriptors
+				)
 				return setState({
 					fileSettings: {
 						edges: getMergedEdges(visibleFiles, withUpdatedPath),
 						markedPackages: getMergedMarkedPackages(visibleFiles, withUpdatedPath),
 						blacklist: getMergedBlacklist(visibleFiles, withUpdatedPath),
-						attributeTypes: getMergedAttributeTypes(allAttributeTypes)
+						attributeTypes: getMergedAttributeTypes(allAttributeTypes),
+						attributeDescriptors: getMergedAttributeDescriptors(allAttributeDescriptors)
 					}
 				})
 			})
 		)
 	)
+
+	constructor(@Inject(ActionsToken) private actions$: Actions, @Inject(State) private state: State) {}
```

Comment:
This did not change right? I would keep it at the top because this would be more intuitive IMHO.